src/main/java/io/github/nucleuspowered/nucleus/modules/fun/commands/SkullCommand.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.modules.fun.commands;
+
+import io.github.nucleuspowered.nucleus.Util;
+import io.github.nucleuspowered.nucleus.internal.annotations.NoCooldown;
+import io.github.nucleuspowered.nucleus.internal.annotations.NoCost;
+import io.github.nucleuspowered.nucleus.internal.annotations.NoWarmup;
+import io.github.nucleuspowered.nucleus.internal.annotations.Permissions;
+import io.github.nucleuspowered.nucleus.internal.annotations.RegisterCommand;
+import io.github.nucleuspowered.nucleus.internal.command.OldCommandBase;
+import io.github.nucleuspowered.nucleus.internal.permissions.PermissionInformation;
+import io.github.nucleuspowered.nucleus.internal.permissions.SuggestedLevel;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.args.GenericArguments;
+import org.spongepowered.api.command.spec.CommandSpec;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.type.SkullTypes;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.text.Text;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@RegisterCommand({"hat"})
+@NoCooldown
+@NoWarmup
+@NoCost
+@Permissions
+public class SkullCommand extends OldCommandBase<Player> {
```

Comment:
OK, apologies from me here again - in order to be able to generate documentation automatically, I updated the `CommandBase`. I explained it in #98, but the other major change is that `createSpec` gets replaced by `getArguments`, which just returns an array of arguments. This allows me to generate documentation using a generator.

The `OldCommandBase` is just so I didn't need to convert all 109 commands at once...

Sorry! Shouldn't be too hard a change though!